pkg/providers/tinkerbell/create.go

```diff
@@ -152,8 +152,10 @@ func (p *Provider) SetupAndValidateCreateCluster(ctx context.Context, clusterSpe
 		if err != nil {
 			return fmt.Errorf("getting TinkerbellIP of management cluster: %s", err)
 		}
-
-		p.datacenterConfig.Spec.TinkerbellIP = managementDatacenterConfig.Spec.TinkerbellIP
+		// Checking for empty first as that returns a different error in the datacenter config validate method below
+		if p.datacenterConfig.Spec.TinkerbellIP != "" && p.datacenterConfig.Spec.TinkerbellIP != managementDatacenterConfig.Spec.TinkerbellIP {
```

Comment:
Do we have defaulting somewhere the populates the value if its empty?

Reply:
Not that I am aware of no, but we were doing this defaulting here below so I would save that until we have other defaulting.